values.yaml

```diff
@@ -499,6 +508,53 @@ connectInject:
   #     namespace-label: label-value
   namespaceSelector: null
 
+  # k8sAllowNamespaces is a list of k8s namespaces to allow Connect sidecar
+  # injection in. If a k8s namespace is not included or is listed in `k8sDenyNamespaces`,
+  # pods in that k8s namespace will not be injected even if they are explicitly
+  # annotated. Use ["*"] to automatically allow all k8s namespaces.
+  #
+  # For example, ["namespace1", "namespace2"] will only allow pods in the k8s
+  # namespaces `namespace1` and `namespace2` to have Connect sidecars injected
+  # and registered with Consul. All other k8s namespaces will be ignored.
+  # Note: `k8sDenyNamespaces` takes precedence over values defined here and
+  # `namespaceSelector` takes precedence over both since it is applied first.
+  # Requires consul-k8s v0.12+
+  k8sAllowNamespaces: ["*"]
+
+  # k8sDenyNamespaces is a list of k8s namespaces that should not allow Connect
+  # sidecar injection. This list takes precedence over `k8sAllowNamespaces`.
+  # `*` is not supported because then nothing would be allowed to be injected.
+  #
+  # For example, if `k8sAllowNamespaces is `["*"]` and k8sDenyNamespaces is
+  # `["namespace1", "namespace2"]`, then all k8s namespaces besides "namespace1"
+  # and "namespace2" will be injected.
+  # Note: `namespaceSelector` takes precedence over this since it is applied first.
+  # Requires consul-k8s v0.12+.
+  k8sDenyNamespaces: ["kube-system", "kube-public"]
```

Comment:
I think unlike in the catalog sync that this should never be injected into and that this should also be hard coded in the injector. This would totally break kube and I wonder how you'd fix it.

Reply:
These are never injected:

Reply:
I think these should be empty but in the comment you can say that kube-system and kube-public are never injected.

Reply:
Removed. Added comment that they will never be injected.